File: lab 4.1/input.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cstdlib>
#include <ctime>

using namespace std;

void generateCase() {
    // Set the sizes of the arrays
    vector<int> ns = {1000, 5000, 10000, 50000, 100000};
    int t = ns.size();
    
    // Open the output file
    ofstream file("input1.txt");
    if (file.is_open()) {
        file << t << "\n";  // Write the number of test cases
        
        // Seed the random number generator
        srand(time(0));
        
        // Generate each test case
        for (int i = 0; i < t; ++i) {
            int n = ns[i];
            file << n << "\n";  // Write the size of the array
            
            // Generate an array of random numbers between 1 and 100000
            vector<int> array(n);
            for (int j = 0; j < n; ++j) {
                array[j] = rand() % 100000 + 1;
            }
            
            // Write the array to the file
            for (int j = 0; j < n; ++j) {
                file << array[j] << " ";
            }
            file << "\n";
            
            // Choose a random element 'x' from the array
            int x = array[rand() % n];
            
            // Randomly select either 'x' or another number 'y'
            int y = rand() % 100000 + 1;
            int z = rand() % 100 + 1;  // Random number between 1 and 100
            file << (z <= 70 ? x : y) << "\n";
        }
    }
    file.close();
}

int main() {
    generateCase();
    cout << "Test cases generated and saved to 'input1.txt'" << endl;
    return 0;
}

File: lab 4.1/input.py
import random  
  
def generateCase():  
  
    ns = [1000,5000,10000,50000,100000]  
    t = len(ns)   
    with open("input1.txt", "w") as file:  
  
        file.write(f"{t}\n")  
        for i in range(t):  
            n = ns[i]  
            file.write(f"{n}\n")  
            array = [random.randint(1, int(1e5)) for _ in range(n)]  
            file.write(" ".join(map(str, array)) + "\n")  
            x = random.choice(array)  
            y = random.randint(1,int(1e5))  
            z = random.randint(1,100)  
            file.write(f"{x if z<=70 else y}\n")  
  
generateCase()  
  
print("Test cases generated and saved to 'test_cases.txt'")